content/wiki/features/Distributed-Content.md

```diff
@@ -0,0 +1,133 @@
+
+- _node_ A requests authentication for _node_ B from _node_ C
+- _node_ B is redirected for log in to _node_ C
+- _node_ C asks for authentication from _node_ B
+- _node_ B authenticates
+- _node_ C verifies _node_ A that _node_ B is _B@C_
+
+(note: in this example A is a server, B is a client, and C is an __authentication provider__, where A and C could be identical)
+
+This allows for secure authentication using widely adopted standards (SASL, SSL) where implementations exist. The SSL certificates in place should come from Let's Encrypt.
+
+### Scenario: user uploads a map
+
+Ok, quite a lot of theory, but how would that look like on a practical example:
+
+- A opens his game
+- A contacts the seed nodes and gets a list of other existing nodes
+- A asks all nodes he knows about metadata based on his popularity factors
+- Since A is a client and does not share any metadata, there is no more interaction with the network
+- Now, ideally, A has a list of servers and current content
+- A decides to go into coop on server B
+- A makes a map on server B and would like to share it
+- A logs into server B and is now logged in as A@B (e.g fohlen@inexor.org)
+- A tells B that he wants the current map to be uploaded
+- B creates metadata
+- More people use the map created by A
+- It gets spread from server B to more servers
+- Hooray"
+
+The scenario for servers is similliar, however not identical:
+
+- A hosts a server
+- If A wants to offer authentication he also needs an SSL certificate
+    - gets a valid FQDN
+    - gets a SSL certificate
+- A starts a server
+    - server contacts seed nodes telling them that they want to be part of the network
+    - fetches list of other servers
+    - from all servers fetches metadata _and_ content based on popularity factors
+- A can now use the server
+- A connects with a client to another server B and authenticates as A@A
+- A uses a texture of his own _server_ on B by using texture@A
+
+#### Drawbacks to this approach
+
+This approach also has some drawbacks:
+
+- even though there are some libraries like libp2p, until everything is set up properly, its still an experiment
+- requiring SSL from every __authentication provider__ is a bit of a setup burden
+    - but makes the whole network much more secure
+- propagating content may take a while
+    - everyone has different settings what they want to download and what they want to share
+    - "unpopular" content may not at all be distributed in the network
+    - however it is still available at the server holding the content = good enough balance
+    - for e.g coop servers would need different settings so new content is propagated faster, but should be doable
+- maintaining the network is virtually impossible
+    - we control the seed nodes but everything beyond that we cannot control
+    - we have to trust our users not to put illegal stuff in the network
```

Comment:
Why is there no way for us to do post-authoring? We could still have the seed nodes to take responsibility and disallow spreading of content. I wouldn't care if one server hosts illegal things, but it shouldn't be able to spread, if we ban the content ID.

Reply:
If a user deletes their own content, it should also propagate through the entire network.

Reply:
For the sake of doing so using a blacklist would be feasible. Anyhow, since the software is open source, nodes __may__ use that blacklist, or the may not.

Reply:
👍 makes sense! I understand that this flexibility also comes with its drawbacks :)
Blacklisting sounds like a good idea. This could also help to free up resources and re-balance the ecosystem.